### PR TITLE
Add zabbix integration

### PR DESCRIPTION
**Pull request by KosukeKamiya**
### What does this PR do?

Add resources of zabbix integration.

### Motivation

Want to move resources from my personal repo to a branch of Datadog/integrations-extras.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

This codes are tested in Zabbix 5.0. But writing test codes are not completed. 

This PR aims to move code from my personal repo to a Datadog/integrations-extras

### CODE REVIEW

File: zabbix/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Agent Check: zabbix
+
+## Overview
+
+This check monitors [zabbix][1] through the Datadog Agent.
```
Comment (ChristineTChen) [inline zabbix/README.md:5]:
```suggestion
This check monitors [Zabbix][1] through the Datadog Agent.
```

Fix integration name capitalization

File: zabbix/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Agent Check: zabbix
```
Comment (ChristineTChen) [inline zabbix/README.md:1]:
```suggestion
# Agent Check: Zabbix
```

File: zabbix/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+This check monitors [zabbix][1] through the Datadog Agent.
+
+## Setup
+
+Follow the instructions below to install and configure this check for an Agent running on a host.
+
+### Installation
+
+To install the zabbix check on your host:
```
Comment (ChristineTChen) [inline zabbix/README.md:13]:
```suggestion
To install the Zabbix check on your host:
```

File: zabbix/assets/configuration/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+name: zabbix
+files:
+- name: zabbix.yaml
+  options:
+  - template: init_config
+    options:
+    - template: init_config/default
+  - template: instances
+    options:
```
Comment (ChristineTChen) [inline zabbix/assets/configuration/spec.yaml:9]:
```suggestion
    options:
    - name: zabbix_user
      description: Username to use Zabbix API.
      required: true
      value:
        type: string
```

Let's add the other options included in the conf.yaml.example here. For more information on config specs, see our developer docs https://datadoghq.dev/integrations-core/meta/config-specs/

Once the options are added, you can automatically update the conf.yaml.example file via `ddev -x validate config --sync`

File: zabbix/assets/configuration/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    - template: instances/default
```
Comment (ChristineTChen) [inline zabbix/assets/configuration/spec.yaml:10]:
```suggestion
    - template: instances/http
    - template: instances/default
```

If this integration uses the HTTP wrapper (wrapper around the Requests library), this config template will automatically support other requests options like (tls verify, ca cert, etc).

File: zabbix/manifest.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  "guid": "bf1fa08e-3df3-40b7-ab1d-1ba685c3057d",
+  "support": "contrib",
+  "supported_os": [
+    "linux",
+    "mac_os",
+    "windows"
+  ],
+  "public_title": "zabbix",
+  "categories": [
+    ""
```
Comment (ChristineTChen) [inline zabbix/manifest.json:19]:
```suggestion
    "network",
    "monitoring"
```

File: zabbix/manifest.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
+  "display_name": "zabbix",
```
Comment (ChristineTChen) [inline zabbix/manifest.json:2]:
```suggestion
  "display_name": "Zabbix",
```

File: zabbix/manifest.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  "maintainer": "KosukeKamiya@users.noreply.github.com",
+  "manifest_version": "1.0.0",
+  "name": "zabbix",
+  "metric_prefix": "zabbix.",
+  "metric_to_check": "",
+  "creates_events": false,
+  "short_description": "",
```
Comment (ChristineTChen) [inline zabbix/manifest.json:9]: Let's include a brief description of this integration

File: zabbix/manifest.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
+{
+  "display_name": "zabbix",
+  "maintainer": "KosukeKamiya@users.noreply.github.com",
+  "manifest_version": "1.0.0",
+  "name": "zabbix",
+  "metric_prefix": "zabbix.",
+  "metric_to_check": "",
```
Comment (ChristineTChen) [inline zabbix/manifest.json:7]: Are there any default metrics that are collected? `metric_to_check` allows the integration tile to get marked as installed when successfully configured.

**Review by ChristineTChen — CHANGES_REQUESTED** [review]
Hi @KosukeKamiya, thanks for opening this PR! Could you please add the integration and yourself to the CODEOWNERS file: https://github.com/DataDog/integrations-extras/blob/master/.github/CODEOWNERS

On a side note, I see that this integration is making requests to the Zabbix API. Our base package supports an HTTP wrapper that can be used. This allows proxy configurations and other validation options. You can read more here: https://datadoghq.dev/integrations-core/base/http

Ive also left some comments on how to use the config spec.yaml file. It provides an ootb way to include default config options (like service + tags) and standardizes the conf.yaml.example format.

By the way, this PR is not being merged to the `master` branch of this repo. Is that intentional?

File: zabbix/manifest.json (hunk 9 above)
Comment (KosukeKamiya) [reply zabbix/manifest.json:7]: There is no default metrics. 

This integration only collects the metrics defined in conf.yaml.

File: zabbix/manifest.json (hunk 8 above)
Comment (KosukeKamiya) [reply zabbix/manifest.json]: Added!

File: zabbix/assets/configuration/spec.yaml (hunk 5 above)
Comment (KosukeKamiya) [reply zabbix/assets/configuration/spec.yaml:49]: I changed the code to use HTTP wrapper, and I added `- template: instances/http` as suggested.

**Comment by KosukeKamiya** [comment]
Hi @ChristineTChen ,
Thank you for reviewing!

> Could you please add the integration and yourself to the CODEOWNERS file

Added.

>  I see that this integration is making requests to the Zabbix API. Our base package supports an HTTP wrapper that can be used. This allows proxy configurations and other validation options

I changed the code to use an HTTP wrapper. Also, I modified configuration files as you suggested. Thank you for your suggestions!

> this PR is not being merged to the master branch of this repo. Is that intentional?

Yes. This is intentional.

I do so because I think this integration doesn't have enough [test codes](https://github.com/KosukeKamiya/integrations-extras/tree/master/zabbix/tests) yet.

Currently, this code is on my personal repository. Since my last day at Datadog is this Friday, I want to put it on Datadog/integrations-extras before that. I understand it should be merged to the master branch with sufficient test codes, but I couldn't complete it due to a lack of time and coding skills.

I'm confident with its behavior itself because it works well in my Zabbix's test environment. Do you think it is okay to merge this integration into the master branch?

**Comment by ChristineTChen** [comment]
Hi @KosukeKamiya, thanks for addressing my comments!

Let's set this PR to merge to master, which will trigger the CI and run validation scripts on your branch.

Since integrations-extras integrations are completely contributor maintained, we would want to get your PR merged to `master`. You can add unit tests using test fixtures if you're unfamiliar with setting up docker e2e test environments. You can see an example of test fixtures used to mimic test responses and asserting metric collection in other integrations, like [envoy](https://github.com/DataDog/integrations-core/pull/5537/files) or the [Istio](https://github.com/DataDog/integrations-core/tree/master/istio/tests) integration.

**Comment by ChristineTChen** [comment]
Hi @KosukeKamiya, since this integration is in -extras can you try running

```
ddev -x validate config --sync
```

The `-x` flag runs validation in the -extras repo, whereas the default is for integrations-core

**Comment by KosukeKamiya** [comment]
@ChristineTChen 
> The -x flag runs validation in the -extras repo, whereas the default is for integrations-core

Thank you for your reply. But I guess it should be `-e` option. Is that right?
```
[ec2-user@ip-172-31-19-46 ~]$ ddev
Usage: ddev [OPTIONS] COMMAND [ARGS]...

Options:
  -c, --core            Work on `integrations-core`.
  -e, --extras          Work on `integrations-extras`.
  -a, --agent           Work on `datadog-agent`.
  -x, --here            Work on the current location.
  --color / --no-color  Whether or not to display colored output (default
                        true).

```

Anyway, I tried both -x and -e option, but the issue was not solved.
```
[ec2-user@ip-172-31-19-46 zabbix]$ ddev -x validate config --sync
Validating default configuration files...
[ec2-user@ip-172-31-19-46 zabbix]$ ddev -e validate config --sync
Validating default configuration files...
Writing config file to `/home/ec2-user/dd/integrations-extras/cyral/datadog_checks/cyral/data/conf.yaml.example`
Writing config file to `/home/ec2-user/dd/integrations-extras/hbase_master/datadog_checks/hbase_master/data/conf.yaml.example`
Writing config file to `/home/ec2-user/dd/integrations-extras/hbase_regionserver/datadog_checks/hbase_regionserver/data/conf.yaml.example`
Writing config file to `/home/ec2-user/dd/integrations-extras/kernelcare/datadog_checks/kernelcare/data/conf.yaml.example`
Writing config file to `/home/ec2-user/dd/integrations-extras/pihole/datadog_checks/pihole/data/conf.yaml.example`
All 46 configuration files are valid!
```

Instead of this, I removed `template: instances/http` from `spec.yaml` on then did sync again on [this commit](https://github.com/DataDog/integrations-extras/pull/722/commits/d63e548c76ee9609583726ca43f557c4d1fe72a6). And the "needs to be synced" issue solved. So I think the cause is from the behavior of sync command or validation logic.

File: zabbix/manifest.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
+{
+  "display_name": "Zabbix",
+  "maintainer": "KosukeKamiya@users.noreply.github.com",
+  "manifest_version": "1.0.0",
+  "name": "zabbix",
+  "metric_prefix": "zabbix.",
+  "metric_to_check": "",
+  "creates_events": false,
+  "short_description": "Collect item histroy through Zabbix API and report them to Datadog as metrics.",
```
Comment (kayayarai) [inline zabbix/manifest.json:9]:
```suggestion
  "short_description": "Collect item history through the Zabbix API and report them to Datadog as metrics.",
```

File: zabbix/datadog_checks/zabbix/data/conf.yaml.example [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    # service: <SERVICE>
+
+## Every instance is scheduled independent of the others.
+#
+instances:
+
+    ## @param zabbix_user - string - required
+    ## Username to use Zabbix API.
+    #
+  - zabbix_user: username
```
Comment (kayayarai) [inline zabbix/datadog_checks/zabbix/data/conf.yaml.example:19]:
```suggestion
  - zabbix_user: <USERNAME>
```

File: zabbix/datadog_checks/zabbix/data/conf.yaml.example [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+    ## @param zabbix_password - string - required
+    ## Password of zabbix_user.
+    #
+    zabbix_password: password
```
Comment (kayayarai) [inline zabbix/datadog_checks/zabbix/data/conf.yaml.example:24]:
```suggestion
    zabbix_password: <PASSWORD>
```

File: zabbix/README.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+## Setup
+
+Follow the instructions below to install and configure this check for an Agent running on a host.
+
+### Installation
+
+To install the Zabbix check on your host:
+
+1. Install the [developer toolkit](https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit) on any machine.
```
Comment (kayayarai) [inline zabbix/README.md:15]:
```suggestion
1. Install the [developer toolkit][2] on any machine.
```

File: zabbix/README.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+### Events
+
+zabbix does not include any events.
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][6].
+
+[1]: https://www.zabbix.com/
```
Comment (kayayarai) [inline zabbix/README.md:48]:
```suggestion
[1]: https://www.zabbix.com/
[2]: https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit
[3]: https://app.datadoghq.com/account/settings#agent
```

File: zabbix/README.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+Follow the instructions below to install and configure this check for an Agent running on a host.
+
+### Installation
+
+To install the Zabbix check on your host:
+
+1. Install the [developer toolkit](https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit) on any machine.
+2. Run `ddev release build zabbix` to build the package.
+3. [Download the Datadog Agent](https://app.datadoghq.com/account/settings#agent).
```
Comment (kayayarai) [inline zabbix/README.md:17]:
```suggestion
3. [Download the Datadog Agent][3].
```

File: zabbix/README.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+## Troubleshooting
+
+Need help? Contact [Datadog support][6].
+
+[1]: https://www.zabbix.com/
+[2]: https://github.com/DataDog/integrations-extras/blob/master/zabbix/datadog_checks/zabbix/data/conf.yaml.example
+[3]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
+[4]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
+[5]: https://github.com/DataDog/integrations-extras/blob/master/zabbix/metadata.csv
+[6]: https://docs.datadoghq.com/help/
```
Comment (kayayarai) [inline zabbix/README.md:53]:
```suggestion
[4]: https://github.com/DataDog/integrations-extras/blob/master/zabbix/datadog_checks/zabbix/data/conf.yaml.example
[5]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
[6]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
[7]: https://github.com/DataDog/integrations-extras/blob/master/zabbix/metadata.csv
[8]: https://docs.datadoghq.com/help/
```

File: zabbix/README.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+To install the Zabbix check on your host:
+
+1. Install the [developer toolkit](https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit) on any machine.
+2. Run `ddev release build zabbix` to build the package.
+3. [Download the Datadog Agent](https://app.datadoghq.com/account/settings#agent).
+4. Upload the build artifact to any host with an Agent and run `datadog-agent integration install -w path/to/zabbix/dist/<ARTIFACT_NAME>.whl`.
+
+### Configuration
+
+1. Edit the `zabbix.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your zabbix performance data. See the [sample zabbix.d/conf.yaml][2] for all available configuration options.
```
Comment (kayayarai) [inline zabbix/README.md:22]:
```suggestion
1. Edit the `zabbix.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your Zabbix performance data. See the [sample zabbix.d/conf.yaml][4] for all available configuration options.
```

File: zabbix/README.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+
+2. [Restart the Agent][3].
```
Comment (kayayarai) [inline zabbix/README.md:24]:
```suggestion
2. [Restart the Agent][5].
```

File: zabbix/README.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+
+### Validation
+
+[Run the Agent's status subcommand][4] and look for `zabbix` under the Checks section.
```
Comment (kayayarai) [inline zabbix/README.md:28]:
```suggestion
[Run the Agent's status subcommand][6] and look for `zabbix` under the Checks section.
```

File: zabbix/README.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+
+## Data Collected
+
+### Metrics
+
+See [metadata.csv][5] for a list of metrics provided by this check.
```
Comment (kayayarai) [inline zabbix/README.md:34]:
```suggestion
See [metadata.csv][7] for a list of metrics provided by this check.
```

File: zabbix/README.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+
+### Service Checks
+
+zabbix does not include any service checks.
```
Comment (kayayarai) [inline zabbix/README.md:38]:
```suggestion
Zabbix does not include any service checks.
```

File: zabbix/README.md [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+### Events
+
+zabbix does not include any events.
```
Comment (kayayarai) [inline zabbix/README.md:42]:
```suggestion
Zabbix does not include any events.
```

File: zabbix/README.md [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][6].
```
Comment (kayayarai) [inline zabbix/README.md:46]:
```suggestion
Need help? Contact [Datadog support][8].
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
A few comments on the documentation, tidying style and linking

**Comment by ChristineTChen** [comment]
HI @KosukeKamiya , once the docs feedback is addressed I can help merge this integration and add testing in a separate PR.

**Comment by KosukeKamiya** [comment]
Hi @kayayarai and @ChristineTChen ,
I addressed the docs feedbacks.

I adopted all suggestions from @kayayarai. 
Since the below error happened, I changed `short_description` on manifest.json because `short_description:'Collect item history through the Zabbix API and report them to Datadog as metrics.' is too long.`

**Review by ChristineTChen — APPROVED** [review]

**Pull request merged**